### DISCUSSION TITLE
Problem in Starter App

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/itwinjs-core#3248] [Q&A] [answered]
I am trying to isolate a bug in my production app. So I created an app from the starter template and I tried to run it and it is giving this error. 
![image](https://user-images.githubusercontent.com/89520756/155074540-a4e6bead-7740-41ca-94b7-7d88adcce233.png)
Please note that I havent changed anything in the starter template. I just added my env variables. 
I have been using starter templates for quite a while now in order to isolate the bugs, I also did it thrice in 3.0 but this problem is new.

**Comment by aruniverse** [comment, marked as answer]
What's the server error? Please look at the network tab, find the failing request, and share the "x-correlation-id" header.

**Reply by hanan-commits-work** [reply]
Hi this is the failing request.
![image](https://user-images.githubusercontent.com/89520756/155270466-c2b59e59-7b5f-4232-abbd-b3e90d88f109.png)
--iModel.getConnectionProps()
and the header is 711dd1f3-1285-4402-b62f-6e1db27c17e9

**Reply by hanan-commits-work** [reply]
I also changed the packages in my production app from 3.0-dev to 3.0 and I received the following error. It was working fine when packages were either 'rc' or 'dev'.
![image](https://user-images.githubusercontent.com/89520756/155271717-75959ee4-f390-4ff3-997a-e883d39800f9.png)

**Reply by aruniverse** [reply]
Looking at your screenshot, you only have 1/3 of the scopes needed to run the viewer. Not sure how it worked for you on the RC version unless you used a really old one. Take a look at the required scopes listed here, https://github.com/iTwin/viewer/blob/master/CHANGELOG-2.0.md

**Reply by hanan-commits-work** [reply]
I changed the scopes too. But it didnt work. Now I am receiving this error.
![image](https://user-images.githubusercontent.com/89520756/155354657-6b7c4a69-dcfa-4ece-a949-d42c84eaf79c.png)

Here is the request that is failing:
![image](https://user-images.githubusercontent.com/89520756/155354733-c6ba16c1-7083-4028-9f73-ceeb815bbca2.png)

And this is the header...
x-correlation-id: 52ad01af-6891-4810-b8c5-9bcd0967eb68

![image](https://user-images.githubusercontent.com/89520756/155354848-a2d666a1-1505-4bbb-ab8f-e6802a0b4232.png)

**Reply by aruniverse** [reply]
I do not see the `imodelaccess:read` scope in your request. Please make sure to update both the client registration on developer.bentley.com and update your src code

**Reply by williamkbentley** [reply]
Also, if you recently added the scopes to your .env file, you'll have to **restart** your create-react-app server. This is because the variables in your .env are essentially _build time_ variables (this has tripped me up before).

**Reply by hanan-commits-work** [reply]
Thank You. It worked. It took a while for scope change to take affect.
